### PR TITLE
readme, tools: add DM-Ansible common operations

### PR DESCRIPTION
**Pull request by CaitinChen**
Via 
https://github.com/pingcap/tidb-tools/pull/95
https://github.com/pingcap/tidb-tools/pull/132
@morgo @lilin90 @csuzhangxc @GregoryIan PTAL

### CODE REVIEW

**Comment by GregoryIan** [comment]
LGTM

File: tools/common-operations-of-dm-ansible.md [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+    ```
+    $ ansible-playbook -i hosts.ini create_users.yml -u root -k
+    ```
+
+    This step creates a `tidb` user on the `172.16.10.74` machine, and configures sudo rules and the SSH mutual trust between the Control Machine and the `172.16.10.74` machine.
+
+### Step 2: Edit the `inventory.ini` file and add the new DM-worker instance
+
+Edit the `inventory.ini` file and add the new DM_worker instance `dm_worker3`.
```
Comment (csuzhangxc) [inline tools/common-operations-of-dm-ansible.md]:
```suggestion
Edit the `inventory.ini` file and add the new DM-worker instance `dm_worker3`. 
```

File: tools/common-operations-of-dm-ansible.md [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+
+```
+$ ansible-playbook rolling_update_monitor.yml --tags=prometheus
+```
+
+## Remove a DM-worker instance
+
+Assuming that you want to remove the `dm_worker3` instance, perform the following steps:
+
+### Step 1: Close the offline DM-worker instance
```
Comment (csuzhangxc) [inline tools/common-operations-of-dm-ansible.md]:
```suggestion
### Step 1: Stop the offline DM-worker instance
```

File: tools/common-operations-of-dm-ansible.md [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+
+2. Run the following command and enter the `root` user password for deploying `172.16.10.80` according to the prompt. 
+
+    ```
+    $ ansible-playbook -i hosts.ini create_users.yml -u root -k
+    ```
+
+    This step creates the `tidb` user account on `172.16.10.80`, configures the sudo rules and the SSH mutual trust between the Control Machine and  the `172.16.10.80` machine.
+
+### Step 2: Close the old DM-master instance
```
Comment (csuzhangxc) [inline tools/common-operations-of-dm-ansible.md]: ditto, and another in L256

File: tools/common-operations-of-dm-ansible.md [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+    username = tidb
+    ```
+
+2. Run the following command and enter the `root` user password for deploying `172.16.10.75` according to the prompt.
+
+    ```
+    $ ansible-playbook -i hosts.ini create_users.yml -u root -k
+    ```
+
+    This step creates the `tidb` user account on `172.16.10.75`, and configures the sudo rules and the SSH mutual trust between the Control Machine and  the `172.16.10.80` machine.
```
Comment (csuzhangxc) [inline tools/common-operations-of-dm-ansible.md]:
```suggestion
    This step creates the `tidb` user account on `172.16.10.75`, and configures the sudo rules and the SSH mutual trust between the Control Machine and  the `172.16.10.75` machine.
```

**Comment by CaitinChen** [comment]
@morgo @csuzhangxc Thank you. PTAL again. @_@

**Comment by csuzhangxc** [comment]
LGTM

**Comment by CaitinChen** [comment]
@lilin90 PTAL

**Comment by CaitinChen** [comment]
@lilin90 PTAL

File: tools/common-operations-of-dm-ansible.md [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+
+### Step 6: Update the dmctl configuration file
+
+```
+ansible-playbook rolling_update.yml --tags=dmctl
+```
+
+## Replace a DM-worker instance
+
+Assuming that you need to maintain the `172.16.10.72` machine or this machine is faulty, and you need to migrate `dm_worker1` from the `172.16.10.72` machine to the `172.16.10.75` machine, perform the following steps:
```
Comment (lilin90) [inline tools/common-operations-of-dm-ansible.md]: Ditto

File: tools/common-operations-of-dm-ansible.md [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+$ ansible-playbook rolling_update_monitor.yml --tags=prometheus
+```
+
+## Replace a DM-master instance
+
+Assuming you need to maintain the `172.16.10.71` machine or this machine is faulty, and you need to migrate the DM-master instance from `172.16.10.71` to `172.16.10.80`, perform the following steps:
+
+### Step 1: Configure the SSH mutual trust and sudo rules on the Control machine
+
+1. Refer to [Configure the SSH mutual trust and sudo rules on the Control Machine](../tools/data-migration-deployment.md#step-5-configure-the-ssh-mutual-trust-and-sudo-rules-on-the-control-machine) and log in to the Control Machine using the `tidb` user account and add `172.16.10.80` to the `[servers]` section of the `hosts.ini` file.
```
Comment (lilin90) [inline tools/common-operations-of-dm-ansible.md]:
```suggestion
1. Refer to [Configure the SSH mutual trust and sudo rules on the Control Machine](../tools/data-migration-deployment.md#step-5-configure-the-ssh-mutual-trust-and-sudo-rules-on-the-control-machine), log in to the Control Machine using the `tidb` user account, and add `172.16.10.80` to the `[servers]` section of the `hosts.ini` file.
```

File: tools/common-operations-of-dm-ansible.md [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+ansible-playbook rolling_update.yml --tags=dmctl
+```
+
+## Replace a DM-worker instance
+
+Assuming that you need to maintain the `172.16.10.72` machine or this machine is faulty, and you need to migrate `dm_worker1` from the `172.16.10.72` machine to the `172.16.10.75` machine, perform the following steps:
+
+### Step 1: Configure the SSH mutual trust and sudo rules on the Control Machine
+
+1. Refer to [Configure the SSH mutual trust and sudo rules on the Control Machine](../tools/data-migration-deployment.md#step-5-configure-the-ssh-mutual-trust-and-sudo-rules-on-the-control-machine) and log in to the Control Machine using the `tidb` user account and add `172.16.10.75` to the `[servers]` section of the `hosts.ini` file.
```
Comment (lilin90) [inline tools/common-operations-of-dm-ansible.md]:
```suggestion
1. Refer to [Configure the SSH mutual trust and sudo rules on the Control Machine](../tools/data-migration-deployment.md#step-5-configure-the-ssh-mutual-trust-and-sudo-rules-on-the-control-machine), log in to the Control Machine using the `tidb` user account, and add `172.16.10.75` to the `[servers]` section of the `hosts.ini` file.
```

File: tools/common-operations-of-dm-ansible.md [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+    username = tidb
+    ```
+
+2. Run the following command and enter the `root` user password for deploying `172.16.10.75` according to the prompt.
+
+    ```
+    $ ansible-playbook -i hosts.ini create_users.yml -u root -k
+    ```
+
+    This step creates the `tidb` user account on `172.16.10.75`, and configures the sudo rules and the SSH mutual trust between the Control Machine and  the `172.16.10.75` machine.
```
Comment (lilin90) [inline tools/common-operations-of-dm-ansible.md]:
```suggestion
    This step creates the `tidb` user account on `172.16.10.75`, and configures the sudo rules and the SSH mutual trust between the Control Machine and the `172.16.10.75` machine.
```

File: tools/common-operations-of-dm-ansible.md [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+    
+### Step 2: Stop the old DM-worker instance
+
+> If the `172.16.10.72` machine is faulty and you cannot log in via SSH, you can ignore this step.
```
Comment (lilin90) [inline tools/common-operations-of-dm-ansible.md]: Ditto

File: tools/common-operations-of-dm-ansible.md [hunk 10]
```diff
@@ -0,0 +1,298 @@
...
+
+```
+$ ansible-playbook stop.yml --tags=dm-worker -l dm_worker1
+```
+
+### Step 3: Edit the `inventory.ini` file and add the new DM-worker instance
+
+Edit the `inventory.ini` file, comment or delete the line where the old `dm_worker1` instance `172.16.10.72` exists, and add the `172.16.10.75` information of the new `dm_worker1` instance.
```
Comment (lilin90) [inline tools/common-operations-of-dm-ansible.md]: Ditto

File: tools/common-operations-of-dm-ansible.md [hunk 11]
```diff
@@ -0,0 +1,298 @@
...
+
+```
```
Comment (lilin90) [inline tools/common-operations-of-dm-ansible.md]:
```suggestion
```ini
```

File: tools/common-operations-of-dm-ansible.md [hunk 12]
```diff
@@ -0,0 +1,298 @@
...
+
+    ```
+    $ ansible-playbook stop.yml --tags=dm-master
+    ```
+
+### Step 3: Edit the `inventory.ini` file and configure the DM-master instance information
+
+Edit the `inventory.ini` file, comment or delete the line where the old DM-master instance exists, and add the information of the new DM-master instance.
+
+```
```
Comment (lilin90) [inline tools/common-operations-of-dm-ansible.md]:
```suggestion
```ini
```

**Comment by CaitinChen** [comment]
@lilian90 Thanks. PTAL again.

File: tools/common-operations-of-dm-ansible.md [hunk 13]
```diff
@@ -0,0 +1,292 @@
...
+
+    2. Run the following command and enter the `root` user password for deploying `172.16.10.80` according to the prompt. 
+
+        ```
+        $ ansible-playbook -i hosts.ini create_users.yml -u root -k
+        ```
+
+        This step creates the `tidb` user account on `172.16.10.80`, configures the sudo rules and the SSH mutual trust between the Control Machine and the `172.16.10.80` machine.
+
+2. Stop the old DM-master instance.
```
Comment (lilin90) [inline tools/common-operations-of-dm-ansible.md]:
```suggestion
2. Stop the DM-master instance that you need to replace.
```

**Comment by CaitinChen** [comment]
@lilin90 *_* Thanks. PTAL again.

File: tools/common-operations-of-dm-ansible.md [hunk 14]
```diff
@@ -0,0 +1,292 @@
...
+
+    2. Use Playbook to download the latest DM binary file and replace the existing binary in the `/home/tidb/dm-ansible/resource/bin/` directory with it automatically.
+
+        ```
+        $ ansible-playbook local_prepare.yml
+        ```
+
+2. Use Ansible to perform the rolling update.
+
+    - Perform a rolling update on the DM-worker instance:
```
Comment (lilin90) [inline tools/common-operations-of-dm-ansible.md]: We should use an ordered list here.

File: tools/common-operations-of-dm-ansible.md [hunk 15]
```diff
@@ -0,0 +1,292 @@
...
+        ansible-playbook rolling_update.yml --tags=dm-master
+        ```
+
+    - Upgrade dmctl:
+
+        ```
+        ansible-playbook rolling_update.yml --tags=dmctl
+        ```
+
+    - Perform a rolling update on DM-worker, DM-master and dmctl
```
Comment (lilin90) [inline tools/common-operations-of-dm-ansible.md]: Add a colon here.

**Pull request merged**